### PR TITLE
feat: player can win card FE

### PR DESCRIPTION
**Pull request by cindyhalim**
**changes**
- refactored game room and player block logic
- tweaked card font sizes to accommodate longer words
- handle card won logic

**demossss**
only send `card_won` event when:

opponent and main player card symbols match :

https://user-images.githubusercontent.com/52426595/162597115-3a3b85c4-96a1-4ec1-94fc-c8af0b9833eb.mov

opponent and main player card symbols match wild card symbols:

https://user-images.githubusercontent.com/52426595/162597172-f6713f3b-5e8a-46e8-b68d-dadd939ad42a.mov

works with multiple matching cards too:

https://user-images.githubusercontent.com/52426595/162597181-8a1dba95-4fbc-4e5f-b55a-cb3a1ab708d5.mov

### CODE REVIEW

File: frontend/src/features/game/game-room.tsx [hunk 1]
```diff
@@ -1,23 +1,19 @@
 import React from "react"
 import { Box, Flex, Text } from "rebass"
-import { CardEmptyState } from "src/components/card-empty-state"
 
+import { CardEmptyState, Card, Button } from "src/components"
 import { useAppSelector } from "src/redux/utils"
 import { theme } from "src/theme"
-import { Card } from "../../components/card"
+import { useGetMainPlayer } from "src/utils/helpers"
```
Comment (emilyhfdong) [inline frontend/src/features/game/game-room.tsx]: should this be `useMainPlayer`?

File: frontend/src/features/game/player-block.tsx [hunk 2]
```diff
@@ -1,29 +1,67 @@
...
+    wildCard,
+  }: {
+    playerCard: IPlayingCard | null
+    mainPlayer: IGamePlayer | undefined
+    wildCard: IWildCard | undefined
+  }) => {
+    const mainPlayerPile = mainPlayer?.playPile
+    const mainPlayerCard = mainPlayerPile?.[mainPlayerPile.length - 1] || null
+
+    const matchesWildCard = wildCard?.symbols.every(
```
Comment (emilyhfdong) [inline frontend/src/features/game/player-block.tsx:44]: o this `every` is new to me! how beautiful

**Review by emilyhfdong — APPROVED** [review]
looks good! just left a baby comment

File: frontend/src/features/game/game-room.tsx (hunk 1 above)
Comment (cindyhalim) [reply frontend/src/features/game/game-room.tsx]: good catch!! 🍤

**Pull request merged**